lib/src/helpers.dart

```diff
@@ -0,0 +1,10 @@
+bool boolFromString(String boolValue) {
+  switch (boolValue?.toLowerCase()) {
+    case 'true':
+      return true;
+    case 'false':
+      return false;
+    default:
+      return null;
```

Comment:
I think it's polemic, but do you think we should thrown here?

Reply:
Yeah, we knew we had to make some kind of decision here. We used `tryParse`to convert the other types from Strings, and it’s behaviour is to emit `null` when the string is not parseable. E.g.
`int.tryParse("two") == null`, `DateTime.tryParse("Xmas") == null`. 

As `bool` does not have a `tryParse` method, we chose to implement something that behaved like the other types do. I believe if we were to throw here, we should throw on unparseable Strings for all types as well, or keep it as it is.

Reply:
I would say that this would be a good place for Dart Non-Nullable types, depending on if the type declaration allows null types, then the current behavior makes sense, but if the user used a strict type than throwing would avoid strange NPE problems further in the flow. I still have to check how dart is doing the backward compatibility of non-nullable types, but maybe it would be a good option here

Reply:
It's especially relevant for distinguishing between required and optional arguments that can be passed as query parameters on deepLink

Reply:
I would say it's ok to keep this way, but we can plan to use the null safety feature in the future

Reply:
I'm okay keeping as it is for now. I agree we should look into supporting null safety, and we can do that, possibly on a major update.